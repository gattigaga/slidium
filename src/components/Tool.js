import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import Icon from "components/Icon";
import Tooltip from "components/Tooltip";

const StyledIcon = styled(Icon)`
  color: ${({ isActive }) => (isActive ? "#222" : "#888")};
  font-size: 20px;
  margin: auto;
`;

const StyledTooltip = styled(Tooltip)``;

const Container = styled.div`
  width: 32px;
  height: 32px;
  box-sizing: border-box;
  border-radius: 4px;
  background: ${({ isActive }) => (isActive ? "#ddd" : "white")};
  margin: 4px;
  display: flex;
  cursor: pointer;
  user-select: none;
  position: relative;

  &:hover {
    background: ${({ isActive }) => (isActive ? "#ddd" : "#eee")};
  }

  &:hover ${StyledIcon} {
    color: #222;
  }

  &:hover ${StyledTooltip} {
    opacity: 1;
  }
`;

const Tool = ({ icon, tooltip, isActive, onClick }) => (
  <Container onClick={onClick} isActive={isActive}>
    {tooltip && <StyledTooltip text={tooltip} />}
    <StyledIcon name={icon} isActive={isActive} />
  </Container>
);

Tool.propTypes = {
  icon: PropTypes.string.isRequired,
  tooltip: PropTypes.string,
  onClick: PropTypes.func,
  isActive: PropTypes.bool
};

Tool.defaultProps = {
  isActive: false
};

export default Tool;
