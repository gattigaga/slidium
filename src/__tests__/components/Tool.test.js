import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Tool from "components/Tool";

let wrapper, onClick;

beforeEach(() => {
  onClick = jest.fn();
  wrapper = shallow(<Tool icon="print" onClick={onClick} />);
});

describe("Tool", () => {
  it("renders without tooltip", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders with tooltip", () => {
    wrapper.setProps({
      tooltip: "Print"
    });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders in active", () => {
    wrapper.setProps({
      isActive: true
    });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("calls 'onClick'", () => {
    expect(onClick.mock.calls.length).toEqual(0);
    wrapper.simulate("click");
    expect(onClick.mock.calls.length).toEqual(1);
  });
});
