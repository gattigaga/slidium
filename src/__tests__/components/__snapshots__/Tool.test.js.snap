// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tool renders in active 1`] = `
<styled.div
  isActive={true}
  onClick={[Function]}
>
  <Styled(Icon)
    isActive={true}
    name="print"
  />
</styled.div>
`;

exports[`Tool renders with tooltip 1`] = `
<styled.div
  isActive={false}
  onClick={[Function]}
>
  <Styled(Tooltip)
    text="Print"
  />
  <Styled(Icon)
    isActive={false}
    name="print"
  />
</styled.div>
`;

exports[`Tool renders without tooltip 1`] = `
<styled.div
  isActive={false}
  onClick={[Function]}
>
  <Styled(Icon)
    isActive={false}
    name="print"
  />
</styled.div>
`;
